nixos/modules/services/networking/oink.nix

```
@@ -80,11 +104,26 @@ in
   };
 
   config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.settings.apiKey != "" || cfg.apiKeyFile != null;
+        message = "services.oink: either apiKey or apiKeyFile must be set";
+      }
+      {
+        assertion = cfg.settings.secretApiKey != "" || cfg.secretApiKeyFile != null;
+        message = "services.oink: either secretApiKey or secretApiKeyFile must be set";
+      }
+    ];
     systemd.services.oink = {
       description = "Dynamic DNS client for Porkbun";
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
-      script = "${cfg.package}/bin/oink -c ${oinkConfig}";
+      script =
+        lib.optionalString (cfg.apiKeyFile != null) "OINK_OVERRIDE_APIKEY=$(cat ${cfg.apiKeyFile}) "
```

```suggestion
        lib.optionalString (cfg.apiKeyFile != null) "OINK_OVERRIDE_APIKEY=\"$(cat ${cfg.apiKeyFile})\" "
```
In a perfect world everybody would provide the API key file in the correct format but in case the file does have spaces or shell statements, we should quote this for safety. 😁